### ISSUE TITLE
Ajax - Unable to update game - Cannot find index of undefined

### ISSUE THREAD

**Issue by artpylon** [ga-wdi-boston/game-project#714]
I'm working on calling the API to update the game following a player selecting a tile. When I run the code, I see an error in console, Cannot find index of undefined. This error is referring to the index in the game array occupied by the newly selected tile. 

I think the issue may be that I'm not referencing then store object and event correctly to successfully define the index for the clicked tile.

I am attempting to store that index like this: each tile is a div that has a dataid that equals it's position in the game array.  For example, the top right tile:

```html
<div class="col-xs-4 col-md-4 box tile" data-id=0>
```
I register the click with an event handler that selects the div which contains all 9 tiles:

```js
$('.tile').on('click', onSelectTile)
```
This event handler calls an event function where I attempt to add the index of the tile to my store object. This function also calls the onSelectTile function which in turn calls the API to update the game.

```js
const onSelectTile = function (event) {
  console.log('this is the tile event ', event)
  console.log('this is the store at select tile ', store)
  event.preventDefault()
  const data = getFormFields(event.target)
  **store.game.cell.index = event.target.dataid**
  turn()
  api.updateGame(data)
    .then(ui.selectTileSuccess)
    .catch(ui.selectTileFailure)
  store.xTurn = !store.xTurn
  console.log(store.xTurn)
}
```
This is the API function, I've successfully curled this format:

```js
const updateGame = function (data) {
  return $.ajax({
    url: config.apiOrigin + '/games' + store.game.id,
    method: 'PATCH',
    headers: {
      Authorization: 'Token token=' + store.userToken
    },
    data: '{"game": {"cell": {"index": store.game.id, "value": "store.game.value"}, "over": false}}'
  })
```
<img width="1255" alt="screen shot 2017-05-16 at 10 22 09 pm" src="https://cloud.githubusercontent.com/assets/26583372/26135827/23ee77ae-3a86-11e7-8c56-1428dd84232c.png">

**Comment by artpylon** [comment]
Also, should mention I've added console logging in the onSelectTile function to see what the store and event contains immediately prior to attempting to store the index.

**Comment by Jcornmanhomonoff** [comment]
Is it making it as far as getting to the console logs in your `onSelectTile` function?

**Comment by jordanallain** [comment]
i don't think you are doing what you think you are with this line

`store.game.cell.index = event.target.dataid`

looking up the .data() jQuery method could prove useful

**Comment by artpylon** [comment]
@jordanallain - it seems like .data() would allow me to add data to an element after it's clicked. But, without attaching something to the clicked tile element, how would I know which index it corresponds to?

**Comment by benjimelito** [comment]
Looks like the issue was that the `event.target` in this case was actually your `section` inside of your `div`, and not the `div` itself. Once you've gotten the `event.target` to be correct, this should be an easier fix.

**Comment by artpylon** [comment]
Resolved several issues here.

1. I had sections inside the tile divs(where the handlers and ids were attached) and the clicks were going to that section instead of the div and thus not getting the id. So, i deleted those section elements.

2. I was referencing the data-id attached to the divs incorrectly, this worked:
```html
<div class="col-xs-4 col-md-4 box tile" data-id=1>
```
```js
store.game.cell.index = this.dataset.id
```
3. I needed to create an empty object in my store.game array before assigning the index value:
```js
store.game.cell = {}
  store.game.cell.index = this.dataset.id
```
